client/asset/eth/eth.go

```diff
@@ -2402,18 +2526,474 @@ func (eth *ETHWallet) checkForNewBlocks(reportErr func(error)) {
 	eth.log.Debugf("tip change: %s (%s) => %s (%s)", prevTip.Number,
 		currentTipHash, bestHdr.Number, bestHash)
 
+	walletList := eth.walletList()
+
 	go func() {
-		for _, w := range eth.walletList() {
+		for _, w := range walletList {
 			w.tipChange(nil)
 		}
 	}()
 	go func() {
-		for _, w := range eth.walletList() {
+		for _, w := range walletList {
 			w.checkFindRedemptions()
 		}
 	}()
 }
 
+// getLatestMonitoredTx looks up a txHash in the monitoredTxs map. If the
+// transaction has been replaced, the latest in the chain of transactions
+// is returned.
+//
+// !!WARNING!!: The latest transaction is returned with its mutex locked.
+// It must be unlocked by the caller. This is done in order to prevent
+// another transaction starting the redemption process before a
+// a potential replacement.
+func (w *assetWallet) getLatestMonitoredTx(txHash common.Hash) (*monitoredTx, error) {
+	w.monitoredTxsMtx.RLock()
+	defer w.monitoredTxsMtx.RUnlock()
+	maxLoops := 100 // avoid an infinite loop in case of a cycle
+	for i := 0; i < maxLoops; i++ {
+		tx, found := w.monitoredTxs[txHash]
+		if !found {
+			return nil, fmt.Errorf("%s not found among monitored transactions", txHash)
+		}
+		tx.mtx.Lock()
+		if tx.replacementTx == nil {
+			return tx, nil
+		}
+		txHash = *tx.replacementTx
+		tx.mtx.Unlock()
+	}
+	return nil, fmt.Errorf("there is a cycle in the monitored transactions")
+}
+
+// recordReplacementTx updates a monitoredTx with a replacement transaction.
+// This change is also stored in the db.
+//
+// originalTx's mtx must be held when calling this funcion.
+func (w *assetWallet) recordReplacementTx(originalTx *monitoredTx, replacementHash common.Hash) error {
+	originalTx.replacementTx = &replacementHash
+	originalHash := originalTx.tx.Hash()
+	if err := w.monitoredTxDB.Store(originalHash[:], originalTx); err != nil {
+		return fmt.Errorf("error recoring replacement tx: %v", err)
```

Comment:
"funcion" and "recoring"

Reply:
Sorry for the silly nits, but will you point a spell checker at the file?  There is still a "funcion" and a "recoring" in another spot.  Potentially user-facing errors are places where I especially would like to eliminate typos.